### DISCUSSION TITLE
CLI: support passing sensitive `Directories`

### DISCUSSION THREAD

**Discussion by sipsma** [dagger/dagger#12421] [Ideas]
Discord ref: https://discord.com/channels/707636530424053791/1120503349599543376/1212120447894818826

Previously, we suggested that users mount sensitive directories in "userspace" from their own code: https://docs.dagger.io/guides/457482/create-app-ci-module/#step-6-add-a-function-to-run-the-application-as-a-local-service

But that no longer is viable when just using `dagger call`.

Off the top of my head thoughts are to either:
1. Create a new core type like `SecretDirectory` or something, which users could accept as an arg to their functions.
2. Somehow modify the existing `Directory` api to support flagging the dir as sensitive, in which case we internally ensure it's never persisted to disk. We'd then also need some syntax from the CLI to allow users to mark directories being passed as sensitive.
   * This is probably my preference provided it's actually viable

**Comment by Nageshbansal** [comment]
@sipsma, Are you working on this Issue? if Not Can I take this one?
